### ISSUE TITLE
Conflicting information in the documentation of the `thinlens` sensor

### ISSUE THREAD

**Issue by maxfrei750** [mitsuba-renderer/mitsuba3#737]
The [docs](https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_sensors.html#perspective-camera-with-a-thin-lens-thinlens) state that the `aperture_radius`, `focus_distance` and `x_fov` are _not_ differentiable. However, when traversing a scene with such a sensor,    they are indeed listed as differentiable:

```
  sensor.near_clip                                                    float ThinLensCamera
  sensor.far_clip                                                     float ThinLensCamera
  sensor.shutter_open                                                 float ThinLensCamera
  sensor.shutter_open_time                                            float ThinLensCamera
  sensor.film.size                                                    ScalarVector2u HDRFilm
  sensor.film.crop_size                                               ScalarVector2u HDRFilm
  sensor.film.crop_offset                                             ScalarPoint2u HDRFilm
  sensor.aperture_radius                                     ∂, D     Float ThinLensCamera
  sensor.focus_distance                                      ∂, D     Float ThinLensCamera
  sensor.x_fov                                               ∂, D     Float ThinLensCamera
  sensor.to_world                                            ∂, D     Transform4f ThinLensCamera
```

Unfortunately, I don't have the time to test it right away. Can you confirm that these properties are differentiable and, if they are, could you tweak the docs?
Thanks!

**Comment by maxfrei750** [comment]
To me, it appears that these properties are indeed _not_ differentiable, which is unfortunate, since it would be cool to be able to optimize them. When using them inside an optimization, their values are constant and their state in the optimizer is just `([0.0], [0.0])`. Still, it would be nice, if one of the maintainers could confirm this, since I might be doing something wrong and don't want others to think that the properties cannot be optimized, while they actually can.

**Comment by njroussel** [comment]
Hi @maxfrei750 

These properties are differentiable, in the sense that the public interface of the `Sensor` will compute values that are properly attached to these parameters. However, I unfortunately don't believe we ever properly supported this in the integrator(s) which must carefully use these gradients.

@Speierers I vaguely remember https://github.com/mitsuba-renderer/mitsuba3/pull/251 and its following commits (after the PR), am I correct in saying that this was never supported/tested?

**Comment by Speierers** [comment]
@njroussel in this PR we only tested the differentiation of the `to_world` parameters of the sensors. I never tried to optimized the aperture radius or FOV parameter.

**Comment by njroussel** [comment]
@Speierers Ok that's exactly what I thought, thank you.

@maxfrei750 I've ran some tests myself, I feel comfortable saying that these parameters are not differentiable with any of the current integrators. In fact, for the the `thinlens` plugin not even its `to_world` parameter can be properly differentiated. I have marked them as such in the source code. We will have to revisit this at some point. Thank you for reporting this.
